src/Invoker.php

```diff
@@ -53,6 +61,19 @@ public function handle(
             $request = ServerRequest::fromGlobals();
         }
 
-        return $this->function->execute($request);
+        try {
+            return $this->function->execute($request);
+        } catch (\Exception $e) {
+            // Log the full error and stack trace
+            ($this->errorLogFunc)((string) $e);
+            // Set "X-Google-Status" to "crash" for Http functions and "error"
+            // for Cloud Events
+            $statusHeader = $this->function instanceof HttpFunctionWrapper
+                ? 'crash'
+                : 'error';
+            return new Response(200, [
```

Comment:
Setting a 200 on an error response seems to defy HTTP RFC and unless Pub/Sub uses this header as an override for HTTP-based retry logic, seems like it will prevent retries.

Reply:
I've updated this to use `500`